SingularityUI/app/models/Request.coffee

```diff
@@ -427,18 +431,22 @@ class Request extends Model
                 return unless confirmed
                 @unpause(confirmed).done callback
 
-    promptRun: (callback) =>
-        try
-            lastCommands = JSON.parse(localStorage.getItem(@localStorageCommandLineInputKeyPrefix + @id))
-        catch e
-            console.error('Could not parse previous commands JSON')
-            lastCommands = []
+    promptRun: (callback, task) => #task is an optional parameter - if it's provided this will rerun it, else this will run a new task
+        if task
+            commands = task.attributes.task.taskRequest.pendingTask.cmdLineArgsList
+        else
+            try
+                lastCommands = JSON.parse(localStorage.getItem(@localStorageCommandLineInputKeyPrefix + @id))
+            catch e
+                console.error('Could not parse previous commands JSON')
+                lastCommands = []
+            commands = if lastCommands? then lastCommands[lastCommands.length - 1] else []
         vex.dialog.prompt
-            message: "<h3>Run Task</h3>"
+            message: "<h3>#{if task then 'Rer' else 'R'}un Task</h3>"
```

Comment:
nothing wrong with this logically, but for the sake of clarity in the code, maybe:

`#{if task then 'Rerun' else 'Run'} Task`